frontend/src/metabase/querying/filters/components/FilterPicker/types.ts

```diff
@@ -1,22 +1,21 @@
 import type * as Lib from "metabase-lib";
 
-export interface FilterPickerWidgetProps {
+export type FilterPickerWidgetProps = {
   query: Lib.Query;
   stageIndex: number;
   column: Lib.ColumnMetadata;
   filter?: Lib.FilterClause;
   isNew: boolean;
-  onChange: (filter: Lib.ExpressionClause) => void;
+  withAddButton: boolean;
+  onChange: (filter: Lib.ExpressionClause, opts: FilterChangeOpts) => void;
   onBack?: () => void;
-}
+};
 
-export interface PickerOperatorOption<Operator> {
-  operator: Operator;
+export type FilterChangeSource = "default" | "add-button";
```

Comment:
In case my suggestion for `FilterChangeOpts` below is rejected, I think we should rename things a bit:

```suggestion
export type FilterChangeSource = "submit" | "submit-and-add-more";
```

We should also adjust `withAddButton` accordingly.

Reply:
Removed this abstraction